services/payment_manager.js

```diff
@@ -11,4 +13,23 @@ exports.process_create_input_req = async(input_response)=>{
 }
 exports.post_create_process = async(req,resp,input_response)=>{
   resp.status(200).send(input_response)
-}
\ No newline at end of file
+}
+
+exports.pre_process_update_payment = async (req) => {
+  return {payment_id:req.params.id, data:req.body}
+}
+exports.process_update_payment = async (input_data) => {
+  const { payment_id, data } = input_data
+  const [affected_rows] = await models.Payment.update(data, {
+    where: {
+        id:payment_id
+    } .then(() => {return models.Enrollment.update({where: {id:req.user_id}})})
```

Comment:
what is updated in the enrollment here ?
also in (id:req.user_id) **id** here is the enrollment id and you  are passing user_id.also req will have only payment id user_id iwill always be taken from the token.from the payment id fetch the plan_id and using this plan_id and user_id find the enrollment id and update the status of the enrollment id